File: server/dotnet/sdk-dotnet-example/Presentation/Models/CreatePayments/CreatePaymentRequest.cs
using Business.Domain.Common;
using Business.Domain.Common.Enums;
using Business.Domain.Enums.Payments;
using Business.Domain.Payments;

namespace Presentation.Models.CreatePayments;

public class CreatePaymentRequest
{
    public decimal? Amount { get; init; }
    
    public Currency? Currency { get; init; }
    
    public PaymentMethodType? Method { get; init; }

    public string? HostedTokenizationId { get; init; }

    public AddressDto? ShippingAddress { get; init; } = new();
    
    public AddressDto? BillingAddress { get; init; } = new();

    public Card? Card { get; init; } = new();

    public Mandate? Mandate { get; init; } = new();
}

File: server/java/src/main/java/com/onlinepayments/example/api/models/payments/CreatePaymentRequest.java
package com.onlinepayments.example.api.models.payments;

import com.onlinepayments.example.service.domain.common.Address;
import com.onlinepayments.example.service.domain.common.enums.Currency;
import com.onlinepayments.example.service.domain.payments.enums.PaymentMethodType;
import com.onlinepayments.example.service.domain.payments.Card;
import com.onlinepayments.example.service.domain.payments.Mandate;

import java.math.BigDecimal;

public record CreatePaymentRequest(
        BigDecimal amount,
        Currency currency,
        PaymentMethodType method,
        String hostedTokenizationId,
        Address shippingAddress,
        Address billingAddress,
        Card card,
        Mandate mandate
) { }
